components/builder-events/src/event.rs

```diff
@@ -0,0 +1,55 @@
+use crate::error::Error;
+use cloudevents::{event::Event,
+                  EventBuilder,
+                  EventBuilderV10};
+use habitat_core::os::net::fqdn;
+use std::fmt;
+use url::Url;
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub struct BuilderEvent {
+    inner:       Event,
+    routing_key: Option<String>,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub enum EventType {
+    PackageChannelMotion,
```

Comment:
Documentation on all of this would be great.

Reply:
I completely agree that documentation will be needed; and in the spirit of short exploratory development iterations and considering all this work is behind a feature flag, I'd like to defer adding documentation until a end-to-end producer->consumer flow is implemented. This will allow us to take any corrective actions and make discoveries validating or invaliding some of our current assumptions about how this architecture design will work, _then_ shore up the documentation :smiley:

Reply:
For sure, it doesn't need to be absolutely complete documentation at this point. Keeping things light and agile at this point is a good idea.

Sometimes, however, I find it useful to add some documentation as I'm fleshing things out (maybe it'd be better to call them "notes" 😅). It can often help to clarify a design in my mind - I can't tell you how many times I've written some code, then started to describe what it's supposed to do in English, only to discover that the code is wrong, could be streamlined in some way, or is just plain unnecessary!

It can also be helpful as a rough guide for reviewers, too!

Reply:
You've convinced me :smiley: I wrote some docs and added them in https://github.com/habitat-sh/builder/pull/1540/commits/7ac0699035e6f015f3b9863dfa72ef4035b9c3ca